controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/UpdateStreamTask.java

```diff
@@ -50,40 +49,25 @@ public UpdateStreamTask(final StreamMetadataTasks streamMetadataTasks,
     public CompletableFuture<Void> execute(final UpdateStreamEvent request) {
         final OperationContext context = streamMetadataStore.createContext(request.getScope(), request.getStream());
 
-        // when update stream event is picked, update task is performed.
-        // 1. check state is updating. else postpone [predicate(time, numberOfAttempts)]
-        // 2. get configuration. If configuration.version == input.version -> perform update
-        // else idempotent check: configuration.version = input.version + 1 && configuration.config == input.configuration.
-        // 3. notify segment store about updated configuration
-        // 4. set state to active.
-
         String scope = request.getScope();
         String stream = request.getStream();
-        StreamConfiguration newConfig = ModelHelper.encode(request.getStreamConfig());
 
-        return streamMetadataStore.getState(scope, stream, true, context, executor)
-                .thenAccept(state -> {
-                    if (!state.equals(State.UPDATING)) {
+        return streamMetadataStore.getConfigurationProperty(scope, stream, true, context, executor)
+                .thenCompose(configProperty -> {
+                    if (!configProperty.isUpdating()) {
                         throw new TaskExceptions.StartException("Update Stream not started yet.");
-                    }
-                })
-                .thenCompose(x ->
-                        checkAndUpdate(request, context, scope, stream, newConfig))
-                .thenCompose(updated -> {
-                    if (updated) {
-                        log.debug("{}/{} updated in metadata store", scope, stream);
-
-                        // we are at a point of no return. Metadata has been updated, we need to notify hosts.
-                        // wrap subsequent steps in retries.
-                        return notifyPolicyUpdate(context, scope, stream, newConfig);
                     } else {
-                        return CompletableFuture.completedFuture(false);
+                        return processUpdate(scope, stream, configProperty, context);
                     }
-                }).thenCompose(x -> completeUpdate(context, scope, stream));
+                });
     }
 
-    private CompletableFuture<Void> completeUpdate(OperationContext context, String scope, String stream) {
-        return FutureHelpers.toVoid(streamMetadataStore.setState(scope, stream, State.ACTIVE, context, executor));
+    private CompletableFuture<Void> processUpdate(String scope, String stream, StreamProperty<StreamConfiguration> configProperty,
```

Comment:
The description of the PR says the following:

```
So the event for truncation and update stream no longer need to contain the update to apply. 
The update is applied by the caller, and if they are able to apply the update successfully, there will be an event posted in requeststream for processing the update. 
```

could you elaborate on what this means? Point to the precise parts of the code you are referring to, please.

Reply:
Ok.